File: largest_even_number_in_string.js
const largestEvenNumber = function (num) {
  for (let i = num.length - 1; i >= 0; i--) {
    if (num[i] % 2 === 0) return num.slice(0, i + 1);
  }
  return "";
};

console.log(largestEvenNumber("32051")); // OutPut: 42021

File: largest_even_number_in_string.py
def largestEvenNumber(num:str) -> str:
    for i in range(len(num)-1,-1,-1):
        if int(num[i]) % 2 == 0:
            return num[:i + 1]
    return ""

print(largestEvenNumber("23231")) #Output = 2323